### PR TITLE
[WIP] Support multiple migration paths

### PR DESCRIPTION
**Pull request by anthonator**
ref https://groups.google.com/forum/#!topic/elixir-ecto/y41IAhOKZZk

Add support for using migrations from multiple paths when migrating.

### CODE REVIEW

**Comment by anthonator** [comment]
This is a WIP but wanted to start getting initial thoughts on `Ecto.Migrator.migrations_paths/1`. `Ecto.Migrator.migrations_paths/1` will return a list of paths specified by the new `:migrations_paths` config option. The `:priv` config option as been removed.

I'll keep working on adding support for providing a list of migration paths down the call chain.

File: lib/ecto/migrator.ex [hunk 1]
```diff
@@ -30,10 +30,24 @@ defmodule Ecto.Migrator do
   """
   @spec migrations_path(Ecto.Repo.t) :: String.t
   def migrations_path(repo) do
+    hd(List.first(migrations_paths(repo)))
```
Comment (josevalim) [inline lib/ecto/migrator.ex]: hd is enough, we don't need both hd and List.first :)

File: lib/ecto/migrator.ex [hunk 2]
```diff
@@ -30,10 +30,24 @@ defmodule Ecto.Migrator do
...
+  end
+
+  @doc """
+  Gets the migrations paths from a repository.
+  """
+  @spec migrations_paths(Ecto.Repo.t) :: [String.t]
+  def migrations_paths(repo) do
+    config = repo.config()
+    config[:migrations_paths] || [default_migration_path(repo)]
```
Comment (josevalim) [inline lib/ecto/migrator.ex]: Let's pass the config as argument to `default_migration_path` to avoid reading it twice.

Comment (anthonator) [reply lib/ecto/migrator.ex]: That's what I was originally doing but

https://github.com/elixir-ecto/ecto/blob/8cb17abc4d9914297b7ca8e3b5760c0ffc4b9fc7/lib/ecto/migrator.ex#L48

requires the repo to build the priv path.

**Comment by josevalim** [comment]
I wrote a comment about this but I think I forgot to send it. :) We have two options:

1. Add an extra_migrations_paths option that is always appended to the default one

2. Or have migrations_paths override the default one if one is given. So if you want to keep the priv one, which is likely, you will have to list it explicitly

**Comment by anthonator** [comment]
Does `:extra_migrations_paths` replace `:migrations_paths`? Might help if I had some examples of the two options.

**Comment by josevalim** [comment]
Ping!

**Comment by josevalim** [comment]
I will close this one as we are moving the migrations code to the ecto_sql repository and this no longer applies to this repo. If there is still an interest on this feature, we can bring it to the new repo. Thanks!

**Pull request closed without merging**